zebra-state/src/service/finalized_state/tests/vectors.rs

```diff
@@ -496,19 +379,156 @@ fn orchard_note_commitment_tree_serialization_pow2() {
     // The purpose of this test is to make sure the serialization format does
     // not change by accident.
     let expected_serialized_tree_hex = "01010178315008fb2998b430a5731d6726207dc0f0ec81ea64af5cf612956901e72f0eee9488053a30c596b43014105d3477e6f578c89240d1d1ee1743b77bb6adc40a0001d3d525931005e45f5a29bc82524e871e5ee1b6d77839deb741a6e50cd99fdf1a";
+
+    orchard_checks(incremental_tree, expected_serialized_tree_hex);
+}
+
+fn sprout_checks(incremental_tree: SproutNoteCommitmentTree, expected_serialized_tree_hex: &str) {
     let serialized_tree = incremental_tree.as_bytes();
+
     assert_eq!(hex::encode(&serialized_tree), expected_serialized_tree_hex);
 
-    let deserialized_tree = OrchardNoteCommitmentTree::from_bytes(&serialized_tree);
+    let deserialized_tree = SproutNoteCommitmentTree::from_bytes(&serialized_tree);
+
+    // Get a legacy deserialized tree from the deserialized tree.
+    let deserialized_legacy_tree = LegacySproutNoteCommitmentTree::from(deserialized_tree.clone());
+
+    // Get a deserialized tree from a legacy deserialized tree.
+    let deserialized_legacy_tree_as_new = deserialized_legacy_tree.into();
+
+    // Check frontiers are the same.
+    incremental_tree.assert_frontier_eq(&deserialized_tree);
+    incremental_tree.assert_frontier_eq(&deserialized_legacy_tree_as_new);
+
+    // Check cached roots are the same.
+    assert_eq!(incremental_tree.root(), deserialized_tree.root());
+    assert_eq!(
+        incremental_tree.root(),
+        deserialized_legacy_tree_as_new.root()
+    );
+
+    // Check recalculated roots are the same
+    assert_eq!(
+        incremental_tree.recalculate_root(),
+        deserialized_tree.recalculate_root()
+    );
+    assert_eq!(
+        incremental_tree.recalculate_root(),
+        deserialized_legacy_tree_as_new.recalculate_root()
+    );
+
+    // Check reclaculated and cached roots are the same
+    assert_eq!(
+        incremental_tree.recalculate_root(),
+        incremental_tree
```

Comment:
We actually need to check the cached root from the serialized format, the one that we construct in the test is guaranteed to be the same.
```suggestion
        deserialised_tree
```

Reply:
https://github.com/ZcashFoundation/zebra/pull/7053/commits/4d05ead48c2505551405d354c05c85ad18868f71